staticd/static_zebra.c

```diff
@@ -921,6 +921,11 @@ extern void static_zebra_request_srv6_sid(struct static_srv6_sid *sid)
 	if (!sid)
 		return;
 
+	if (!sid->locator) {
+		static_zebra_srv6_manager_get_locator(sid->locator_name);
```

Comment:
is not it that we missed the srv6 locator event ?
can you please confirm if the srv6 locators are really populated in static?

Reply:
@cscarpitta , I reformulate.
If staticd did not have access to locator information, this means zebra may not have sent locators information to static.
- either static is not part of the daemons interested in locators
- or a retry needs to be done
- or your patch is fine

please advert,

Reply:
- This protection should be done for all dameons that read SRv6 SID. The same code pattern should be applied to ISIS and BGP.
- A retry mechanism for locator should also be put in place.

Reply:
1- The protection for BGP and ISIS has been already added in https://github.com/FRRouting/frr/pull/18069 and https://github.com/FRRouting/frr/pull/18178.
2- Agree that we should have a mechanism to resend the locator to the protocols. I will do this in a separate PR.